### PR TITLE
Add Implementation of AddAppt in DG

### PR DESCRIPTION
**Pull request by yadobler**
image
image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -156,6 +156,37 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
+   The input command text is parsed to identify the type of command and the target `Person` for the appointment. If the `Person` is found in the model, the process continues with creating an appointment.
+
+2. **Create and Add Appointment:**  
+   A new `Appointment` is created with the provided details, and an updated `Person` object is prepared, associating this appointment with the target individual.
+
+3. **Model Update:**  
+   The model replaces the old `Person` with this modified version in the address book, which then updates the internal list of appointments to include the new entry.
+
+4. **Automatic UI Refresh:**  
+   The `AppointmentListPanel` UI component, which observes changes in the list of appointments, detects the addition and refreshes its display. The UI then reflects this change by showing a new `AppointmentCard` for the recently added appointment.
```
Comment (yong-zhuo) [inline docs/DeveloperGuide.md:177]: Good explanation of how the UI changes when the Model changes 👍🏻

File: docs/diagrams/AddApptCommandSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+box Logic LOGIC_COLOR_T1
+participant "LogicManager" as Logic LOGIC_COLOR
+participant "AddApptCommand" as AddApptCommand LOGIC_COLOR
+participant "CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "Model" as Model MODEL_COLOR
+participant "AddressBook" as AddressBook MODEL_COLOR
```
Comment (yong-zhuo) [inline docs/diagrams/AddApptCommandSequenceDiagram.puml:15]: I believe we would need to put `:` beside all the class name 

image

File: docs/images/AddApptCommandSequenceDiagram.png
Comment (yong-zhuo) [inline docs/images/AddApptCommandSequenceDiagram.png:1]: Good sectioning of Logic/Model/UI

File: docs/diagrams/AddApptCommandSequenceDiagram.puml (hunk 2 above)
Comment (PresleyChew) [reply docs/diagrams/AddApptCommandSequenceDiagram.puml:15]: Yes, unless we are calling static method, then we call from class itself. Else should have `:`
image

**Review by PresleyChew — COMMENTED** [review]
Overall LGTM ensure sequence diagram syntax is correct and ready for merge

**Pull request merged**